tdrs-backend/scripts/db_backup.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/python
+
+"""Back up the database to S3 bucket."""
+# Need to run "export LD_LIBRARY_PATH=~/deps/1/python/lib/" before running python from ~/deps/1/python
+# which has boto3 available
+
+import subprocess
+import os
+import boto3
+import logging
+import json
+import sys
+import getopt
+
+OS_ENV = os.environ
+
+try:
+    SPACE = json.loads(OS_ENV['VCAP_APPLICATION'])['space_name']
+
+    # Postgres client pg_dump directory
+    f = subprocess.Popen(["find", "/", "-iname", "pg_dump"], stdout=subprocess.PIPE)
+    f.wait()
+    output, error = f.communicate()
+    output = output.decode("utf-8").split('\n')
+    if output[0] == '':
+        raise Exception("Postgres client is not found")
+
+    POSTGRES_CLIENT = None
+    for _ in output:
+        if 'pg_dump' in str(_) and 'postgresql' in str(_):
+            POSTGRES_CLIENT = _[:_.find('pg_dump')]
+
+    S3_ENV_VARS = json.loads(OS_ENV['VCAP_SERVICES'])['s3']
+    S3_CREDENTIALS = S3_ENV_VARS[0]['credentials']
+    S3_ACCESS_KEY_ID = S3_CREDENTIALS['access_key_id']
+    S3_SECRET_ACCESS_KEY = S3_CREDENTIALS['secret_access_key']
+    S3_BUCKET = S3_CREDENTIALS['bucket']
+    S3_REGION = S3_CREDENTIALS['region']
+    DATABASE_URI = OS_ENV['DATABASE_URL']
+
+    # Set AWS credentials in env, Boto3 uses the env variables for connection
+    os.environ["AWS_ACCESS_KEY_ID"] = S3_ACCESS_KEY_ID
+    os.environ["AWS_SECRET_ACCESS_KEY"] = S3_SECRET_ACCESS_KEY
+
+    # Set Database connection info
+    AWS_RDS_SERVICE_JSON = json.loads(OS_ENV['VCAP_SERVICES'])['aws-rds'][0]['credentials']
+    DATABASE_PORT = AWS_RDS_SERVICE_JSON['port']
+    DATABASE_PASSWORD = AWS_RDS_SERVICE_JSON['password']
+    DATABASE_DB_NAME = AWS_RDS_SERVICE_JSON['db_name']
+    DATABASE_HOST = AWS_RDS_SERVICE_JSON['host']
+    DATABASE_USERNAME = AWS_RDS_SERVICE_JSON['username']
+
+    # write .pgpass
+    with open('/home/vcap/.pgpass', 'w') as f:
+        f.write(DATABASE_HOST+":"+DATABASE_PORT+":"+DATABASE_DB_NAME+":"+DATABASE_USERNAME+":"+DATABASE_PASSWORD)
+    os.environ['PGPASSFILE'] = '/home/vcap/.pgpass'
+
+except Exception as e:
+    print(e)
+    sys.exit(1)
+
+
+def backup_database(file_name,
+                    postgres_client,
+                    database_uri):
+    """Back up postgres database into file."""
+    """
+    :param file_name: back up file name
+    :param postgres_client: directory address for postgres application
+    :param database_uri: postgres URI
+    pg_dump -F c --no-acl --no-owner -f backup.pg postgresql://${USERNAME}:${PASSWORD}@${HOST}:${PORT}/${NAME}
+    """
+    try:
+        os.system(postgres_client + "pg_dump -Fc --no-acl -f " + file_name + " -d " + database_uri)
+        return True
+    except Exception as e:
+        logging.log(e)
```

Comment:
Is formatting ever setup for logging? I see a mis-match between using `print(e)` and `logging.log(e)` throughout this script.

Reply:
background: print(e) are for errors not important. logging.log(e) log the actual errors.

Reply:
Ran into this:

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/vcap/app/scripts/db_backup.py", line 242, in <module>
    main(sys.argv[1:])
  File "/home/vcap/app/scripts/db_backup.py", line 238, in main
    handle_args(argv=argv)
  File "/home/vcap/app/scripts/db_backup.py", line 202, in handle_args
    logging.log(e)
TypeError: log() missing 1 required positional argument: 'msg'
```